File: gameSrc/model/classes/tile.js
//
//
// @class
//
cwt.TileClass = my.Class({

  constructor: function() {
    this.type = null;
    this.unit = null;
    this.property = null;
    this.visionTurnOwner = 0;
    this.variant = 0;
    this.visionClient = 0;
  },

  //
  //
  // @return {boolean}
  //
  isOccupied: function() {
    return this.unit !== null;
  },

  //
  //
  // @return {boolean}
  //
  isVisible: function() {
    return this.visionTurnOwner > 0;
  }

});

File: gameSrc/net/wolfTec/model/Tile.java
package net.wolfTec.model;

import net.wolfTec.database.TileType;
import org.stjs.javascript.annotation.Namespace;

@Namespace("cwt")
public class Tile {

    public Unit unit;
    public Property property;
    public TileType type;

    public int visionTurnOwner = 0;
    public int variant = 0;
    public int visionClient = 0;

    /**
     *
     * @return true if the property is occupied, else false
     */
    public boolean isOccupied () {
        return this.unit != null;
    }

    public boolean isVisible () {
        return this.visionTurnOwner > 0;
    }
}
